### PR TITLE
add granulate gmaestro addon

### PR DESCRIPTION
**Pull request by kirshenboim**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by shapirov103** [comment]
@kirshenboim I am trying to deploy the pattern and I got stuck on the prometheus and grafana secrets. The downloaded  deployment config does not contain it.

1. Are grafana and prometheus fields mandatory? (I see that your add-on design states that these fields are mandatory, just making sure)
2. The point of the blueprint is to provide an actual full blueprint of the deployment, which means end-to-end solution that works. I don't see add-ons for prometheus or Grafana, but even if they were supplied, it is unclear how customers can supply the secrets ahead of deployment.
3. How did you test your pattern? May be I am missing something basic, please point me to the approach. 

In order to get your expected ROI from this effort, the approach should work for an enterprise customer that has a pipeline provisioning clusters across dev/test/prod and all the regions in fully automated mode (that means no post deployment steps or incremental addition of add-ons). 

Happy to discuss!

**Comment by kirshenboim** [comment]
@shapirov103 ive deleted the grafanaMetricsSecretName, grafanaLogsSecretName

File: lib/gmaestro-construct/index.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        const stackId = `${id}-blueprint`;
+
+        let gmaestroAddOnProps = {
+            namespace: "test", // replace with the namespace from the deployment yaml
+            clientIdSecretName: CLIENT_ID_SECRET_NAME, // replace with a secret name in secrets manager that you have created
+            clientName: "test_client", // replace with the client name from the deployment yaml
+            clusterName: "test_cluster", // replace with the service name from the deployment yaml
+        } as gmaestroAddOn.GmaestroAddOnProps
+
+        const addOns: Array<blueprints.ClusterAddOn> = [
```
Comment (shapirov103) [inline lib/gmaestro-construct/index.ts:21]: minor why not just have it as:

```typescript
const gmaestroAddOn = new gmaestroAddOn.GmaestroAddOn( {
    clusterName: ...
});

blueprints.EksBlueprint.builder()
    ...
    .addOns(gmaestroAddOn)
    ...
```

**Review by shapirov103 — COMMENTED** [review]
@kirshenboim looks better, thank you for the update. I added one minor comment. 

I was going through the deployment instructions and your documentation states: Check that metrics server is running: `kubectl get pods --all-namespaces | grep metrics-server`

I assume that means metrics server should be deployed. The expectation is that the blueprint is going to be fully functional. 

If your documentation is correct, please add `MetricsServerAddoOn` to the blueprint (https://aws-quickstart.github.io/cdk-eks-blueprints/addons/metrics-server/). 
If there are any dependencies please add those as well.
You can also add a dependency on MetricsServer to prevent deployment of Blueprints without metrics server addon, Please see this section in the guide: https://aws-quickstart.github.io/cdk-eks-blueprints/extensibility/#add-on-dependencies

File: lib/gmaestro-construct/index.ts (hunk 1 above)
Comment (fespigares) [reply lib/gmaestro-construct/index.ts:21]: @kirshenboim been trying to deploy myself, and I'm unable to identify the proper value for "clientName" addon property. The README.MD on your NPM package mentions copying the value from 'prometheus.configs.scrape_configs.static_configs.labels.client_name', but that property does not exists on the YAML file automatically generated by gMaestro web interface. Is this value still relevant?

Comment (kirshenboim) [reply lib/gmaestro-construct/index.ts:21]: you are right ...fixed

**Comment by kirshenboim** [comment]
@shapirov103 ive fixed your comments (added dependency MetricsServerAddOn ,deleted the client_name)

**Comment by fespigares** [comment]
Hi @kirshenboim, while testing your last commit, I am facing this error while trying to deploy your addon:

```
❯ cdk deploy
DEBUG Core add-on vpc-cni is at version v1.12.2-eksbuild.1
DEBUG Core add-on coredns is at version v1.8.7-eksbuild.4
DEBUG Core add-on kube-proxy is at version v1.24.7-eksbuild.2
/Users/fespigares/Documents/dev/granulate-gmaestro/node_modules/@aws-quickstart/eks-blueprints/lib/utils/addon-utils.ts:35
      addOns.forEach( (addOn) => {
             ^
AssertionError [ERR_ASSERTION]: Missing a dependency for MetricsServerAddOn for eks-blueprint
    at /Users/fespigares/Documents/dev/granulate-gmaestro/node_modules/@aws-quickstart/eks-blueprints/lib/utils/addon-utils.ts:37:9
    at Array.forEach (<anonymous>)
    at GmaestroAddOn.descriptor.value (/Users/fespigares/Documents/dev/granulate-gmaestro/node_modules/@aws-quickstart/eks-blueprints/lib/utils/addon-utils.ts:35:14)
    at new EksBlueprint (/Users/fespigares/Documents/dev/granulate-gmaestro/node_modules/@aws-quickstart/eks-blueprints/lib/stacks/eks-blueprint-stack.ts:263:34)
    at BlueprintBuilder.build (/Users/fespigares/Documents/dev/granulate-gmaestro/node_modules/@aws-quickstart/eks-blueprints/lib/stacks/eks-blueprint-stack.ts:195:16)
    at Object.<anonymous> (/Users/fespigares/Documents/dev/granulate-gmaestro/bin/granulate-gmaestro.ts:33:6)
    at Module._compile (node:internal/modules/cjs/loader:1275:14)
    at Module.m._compile (/Users/fespigares/Documents/dev/granulate-gmaestro/node_modules/ts-node/src/index.ts:1618:23)
    at Module._extensions..js (node:internal/modules/cjs/loader:1329:10)
    at Object.require.extensions.<computed> [as .ts] (/Users/fespigares/Documents/dev/granulate-gmaestro/node_modules/ts-node/src/index.ts:1621:12) {
  generatedMessage: false,
  code: 'ERR_ASSERTION',
  actual: undefined,
  expected: true,
  operator: '=='
}

Subprocess exited with error 1
```
as we have discussed today, MetricsServerAddOn is already deployed in the cluster I'm trying to add gMaestroaddOn to.
@shapirov103 , maybe you can suggest a fix?

**Comment by shapirov103** [comment]
@fespigares the pattern looks good, I suspect your problem may be related to build: this repo requires to run `make build` before deploying the patterns. 
Since the pattern was modified to include the metrics server, you need to rebuild:

```
make build
```

**Comment by shapirov103** [comment]
Can we get assurance on support of this add-on and pattern if we merge it? I understand the main committer has left the company, however if we publish something we need continuity.

**Comment by noamsalinger** [comment]
@shapirov103 Yes we have a support team on call.
Yes I am personally committed to supporting any future development and assistance for the addon

**Comment by elamaran11** [comment]
@noamsalinger We dont see any traction on this pattern and i also see `conflicts` which needs to be resolved. Could you please check on this, fix all that is required so we can move forward with this pattern.?

**Comment by elamaran11** [comment]
Folks any updates on this PR?

**Pull request closed without merging**